### PR TITLE
Optimized code

### PR DESCRIPTION
**Pull request by 18JE0782**
#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1364"></a>

### CODE REVIEW

File: dynamic_programming/kadane.cpp [hunk 1]
```diff
@@ -19,11 +20,12 @@ int main() {
...
+    for (i = 0; i < n; i++) 
+    {
         std::cin >> a[i];
     }
-    int max_sum = maxSubArraySum(a, n);
-    std::cout << "Maximum contiguous sum is " << max_sum;
+    std::cout << "Maximum contiguous sum is " << maxSubArraySum(a);;
     return 0;
 }
+
```
Comment (Panquesito7) [inline dynamic_programming/kadane.cpp:31]:
```suggestion
```

File: dynamic_programming/kadane.cpp [hunk 2]
```diff
@@ -1,9 +1,10 @@
-#include <climits>
 #include <iostream>
-
-int maxSubArraySum(int a[], int size) {
+#include <climits>
+#include<vector>
```
Comment (Panquesito7) [inline dynamic_programming/kadane.cpp:3]:
```suggestion
#include <vector>
```

**Review by Panquesito7 — APPROVED** [review]

**Review by Panquesito7 — APPROVED** [review]

**Comment by 18JE0782** [comment]
@Panquesito7  @kvedala Please Merge this commit if approved.

**Comment by 18JE0782** [comment]
@ayaankhan98  Please Merge this commit if approved or suggest any changes.

**Review by ayaankhan98 — APPROVED** [review]

**Pull request merged**